### PR TITLE
Select question ui

### PR DESCRIPTION
**Pull request by alexmorris**
- Updated drop zone area
- Moved add section to bottom
- Visual defect fixes
- Added select question UI and angular rework to fix safari issue

### CODE REVIEW

File: author/static/js/q_builder/templates/question_group.html [hunk 1]
```diff
@@ -1,7 +1,7 @@
 <!-- This template is responsible for rendering a group element. It uses
 		 the above list template to render each group column -->
 <script type="text/ng-template" id="group.html">
-    <i class="fa fa-times deleteitem" ng-click="delete($index)"
+    <i class="fa fa-times deleteitem" ng-click="list.splice($index, 1)"
```
Comment (dhilton) [inline author/static/js/q_builder/templates/question_group.html]: This is an issue as the delete function does more than simply delete; we believe this is a merge issue.

Comment (warren-methods) [inline author/static/js/q_builder/templates/question_group.html]: I tried rebasing this to master and this still happens, so we might have to cherry-pick the required changes instead. :-1:

**Comment by warren-methods** [comment]
I've rebased and fixed the conflicts for the second time. It would be nice to get this merged now.

**Comment by dhilton** [comment]
@alexmorris Is this ready for review and merge now?

**Comment by alexmorris** [comment]
nope, on cleanup duty

**Pull request merged**